addon/-private/system/relationships/state/has-many.js

```diff
@@ -143,7 +167,26 @@ export default class ManyRelationship extends Relationship {
       return;
     }
     super.removeInternalModelFromOwn(internalModel, idx);
-    let manyArray = this.manyArray;
+    // TODO: explain this madness
+    this._removeInternalModelFromManyArray(this.manyArray, internalModel, idx);
+    this._removeInternalModelFromManyArray(this._retainedManyArray, internalModel, idx);
```

Comment:
we should maybe do this in the async case?

Reply:
we discussed this OOB, but LMK if you still think there's an issue here